libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -224,9 +224,15 @@ func (r *reporter) snapshotLoop(namespace string, period time.Duration) {
 			continue
 		}
 
+		// Prefix for the inside the event. For backward compatibility stats is renamed to metrics.
+		prefix := namespace
+		if namespace == "stats" {
+			prefix = "metrics"
+		}
```

Comment:
What about passing the prefix as a parameter to this function to keep it generic? Then the calls would be:
```
go r.snapshotLoop("state", "state", c.StatePeriod)
go r.snapshotLoop("stats", "metrics", c.MetricsPeriod)
```

Reply:
good idea, changed.